rlclientlib/logger/logger_facade.cc

```diff
@@ -11,31 +11,55 @@ namespace reinforcement_learning {
 
     cb_logger_facade::cb_logger_facade(const utility::configuration& c, i_message_sender* sender, utility::watchdog& watchdog, i_time_provider* time_provider, error_callback_fn* perror_cb)
     : version(c.get_int(name::PROTOCOL_VERSION, value::DEFAULT_PROTOCOL_VERSION))
-    , v1(version == 1 ? new interaction_logger(c, sender, watchdog, time_provider, perror_cb) : nullptr) {
+    , v1(version == 1 ? new interaction_logger(c, sender, watchdog, time_provider, perror_cb) : nullptr)
+    , v2(version == 2 ? new generic_event_logger(
+      sender,
+      c.get_int(name::INTERACTION_SEND_HIGH_WATER_MARK, 198 * 1024),
+      c.get_int(name::INTERACTION_SEND_BATCH_INTERVAL_MS, 1000),
+      c.get_int(name::INTERACTION_SEND_QUEUE_MAX_CAPACITY_KB, 16 * 1024) * 1024,
+      c.get(name::QUEUE_MODE, "DROP"),
+      watchdog,
+      time_provider,
+      perror_cb) : nullptr) {
     }
 
     int cb_logger_facade::init(api_status* status) {
       switch (version) {
         case 1: return v1->init(status);
+        case 2: return v2->init(status);
         default: return protocol_not_supported(status);
       }
     }
 
-    int cb_logger_facade::log(const char* event_id, const char* context, unsigned int flags, const ranking_response& response, api_status* status, learning_mode learning_mode) {
+    int cb_logger_facade::log(const char* context, unsigned int flags, const ranking_response& response, api_status* status, learning_mode learning_mode) {
       switch (version) {
-        case 1: return v1->log(event_id, context, flags, response, status, learning_mode);
+        case 1: return v1->log(response.get_event_id(), context, flags, response, status, learning_mode);
+        case 2:
+          v2::LearningModeType lmt;
+          RETURN_IF_FAIL(get_learning_mode(learning_mode, lmt, status));
+          return v2->log(response.get_event_id(), serializer.event(context, flags, lmt, response), serializer.type, status);
```

Comment:
yes. Let me share some performance tests results - I do not see any performance drop. Probably because serialization here is not differ that much from extra copying of the data (in v1 code path)

Reply:
![image](https://user-images.githubusercontent.com/41013086/92160781-91644d80-edfd-11ea-94a9-d777157287c6.png)

Reply:
each point - latency averaged by 10000 choose_rank calls.